pkg/status/appregistryerror.go

```diff
@@ -0,0 +1,40 @@
+package status
```

Comment:
Is this defined in the status package to avoid an import cycle? If we aren't using the type for control flow, why not just not import this type at all and in the status reporting methods just use the generic `Error` type?

Reply:
We want to be able to update the `ClusterOperator` with both a `Reason` and a `Message` as described [here](https://github.com/operator-framework/operator-marketplace/pull/235#discussion_r309719019)

Reply:
There is no reason this is in the status package other than the fact that it is used to report status.